GDJS/tests/tests/Extensions/lightruntimeobject.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Tests for Light Object
+ */
+
+describe('gdjs.LightRuntimeObject', function () {
```

Comment:
These are pretty small tests (which don't really add a lot of value), I'll be writing more especially related to raycasting. Can you explain to me how can I make game objects having certain behaviors (light obstacle), I see the behaviors array, but what should it contain? The name of the behavior? (`LightObstacleBehavior`)

Reply:
That's a great start! Definitely keep this test and let's add another one or more with obstacles.

See `platformerobjectruntimebehavior.spec.js` as an example of creating an object with a certain behavior.
The `behaviors` array is an array of `BehaviorData`, which are the objects being passed to the behavior constructor. They contain at least the `name` of the behavior (not important), the `type` (should be set to `Lighting::LightObstacleBehavior` in your case) and the rest is what is expected by your behavior.

You create the new object (probably a raw `gdjs.RuntimeObject` like in `platformerobjectruntimebehavior.spec.js`, then override `getWidth`/`getHeight` to give it some volume, then you can add it to the runtime scene (`runtimeScene.addObject`) and set its position.

I recommend doing this initialization in a separate helper function if you want to reuse it for multiple tests. Then, each test can put a light object at different places and check the vertices/etc :) (it's good practice to make sure that tests are entirely autonomous and don't rely on other tests being run before - don't hesitate to make multiple tests with `describe`/`it` rather than one huge test - multiple tests are much easier to understand and see why they are failing).